### PR TITLE
Copy example Objectives from Mimic Sim to Mujoco, with latest comments

### PR DESCRIPTION
**Pull request by davetcoleman**
Copy in these PRs to new Mujoco config
https://github.com/PickNikRobotics/moveit_studio_ur_ws/pull/308
https://github.com/PickNikRobotics/moveit_studio_ur_ws/pull/307
https://github.com/PickNikRobotics/moveit_studio_ur_ws/pull/306
https://github.com/PickNikRobotics/moveit_studio_ur_ws/pull/305

### CODE REVIEW

**Comment by MikeWrock** [comment]
The mock hardware objectives use the key `mock_hardware_objectives`:
https://github.com/PickNikRobotics/moveit_studio_ur_ws/blob/pr_add_objectives_mujoco/src/picknik_ur_mock_hw_config/config/config.yaml#L27-L32

To add them to the mujoco config without having to duplicate (and maintain two versions of) the objective you can add additional library paths by modifying [these lines](https://github.com/PickNikRobotics/moveit_studio_ur_ws/blob/pr_add_objectives_mujoco/src/picknik_ur_mujoco_config/config/config.yaml#L45-L50):
```
  objective_library_paths:
    # You must use a unique key for each package.
    # The picknik_ur_base_config uses "core"
    mock_hardware_objectives:
      package_name: "picknik_ur_mock_hw_config"
      relative_path: "objectives"
    sim_objectives:
      package_name: "picknik_ur_mujoco_config"
      relative_path: "objectives"
```

**Comment by davetcoleman** [comment]
I don't know how to finish this PR, and do not currently have bandwidth, but I think these basic objectives to belong in our new example robot config package. Could someone finish this PR please?

Note the reason I copied these in was because i added a lot of new descriptions and comments to those objectives, and didn't want them to get forever lost in the mock hw robot config.

File: src/picknik_ur_mujoco_config/objectives/3_waypoint_pick_and_place.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <SubTree ID="Move to Waypoint" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" planner_interface="moveit_default"/>
+      <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
+      <!--Run pick and place on loop-->
+      <Decorator ID="KeepRunningUntilFailure">
+        <Control ID="Sequence">
+          <!--Pick object from left, put it down at the CNC machine, and go back to center pose-->
+          <Control ID="Sequence">
+            <SubTree ID="Move to Waypoint" waypoint_name="Grasp Left" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" planner_interface="moveit_default"/>
+            <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0.7929"/>
+            <SubTree ID="Move to Waypoint" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" planner_interface="moveit_default"/>
```
Comment (dyackzan) [inline src/picknik_ur_mujoco_config/objectives/3_waypoint_pick_and_place.xml:18]: The "Look at Machine" waypoint does not exist outside of the mock sim config

**Review by dyackzan — COMMENTED** [review]
This is definitely not ready to merge, but it sounds like @bgill92 is assigned the issue for this: https://github.com/PickNikRobotics/moveit_studio/issues/8368.

I'll go ahead and note some changes that need to be made here:
* Move these objectives to a package (I think they are supposed to be in arm_on_rail_sim now since the mujoco config was renamed to that)
* Fix the waypoints

**Pull request closed without merging**